src/seedu/addressbook/data/person/Address.java

```diff
@@ -10,23 +10,56 @@
 
     public static final String EXAMPLE = "123, some street";
     public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
+    public static final String MESSAGE_ADDRESS_INVALID = "Person address must be in the following format a/BLOCK, STREET, UNIT, POSTAL_CODE ";
     public static final String ADDRESS_VALIDATION_REGEX = ".+";
 
+    public static final int BLOCK_NUMBER = 0;
+    public static final int STREET_NUMBER = 1;
+    public static final int UNIT_NUMBER = 2;
+    public static final int POSTALCODE_NUMBER = 3;
+    
+    public static final int REQUIRED_NUMBER_OF_ADDRESS_FIELDS = 4;
+    
     public final String value;
+    
+    
+    private final Block block;
```

Comment:
Hmm, just realized that the address may be subject to edits after creation, so it should not be final.